### PR TITLE
Skip explicit AWS load balancer/security group deletion if infrastructure already deleted

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**: If the infrastructure has already been deleted we do not need to explicitly delete AWS load balancers and security groups.

**Release note**:

```improvement operator
The explicit AWS load balancer and security group deletion is skipped if the infrastructure has already been deleted.
```

### CODE REVIEW

File: pkg/operation/terraformer/config.go [hunk 1]
```diff
@@ -68,29 +68,26 @@ func (t *Terraformer) prepare() (int, error) {
 	// Check whether the required ConfigMaps and the Secret exist
 	numberOfExistingResources := 3
```
Comment (timuthy) [inline pkg/operation/terraformer/config.go]: Would make it a constant.

File: pkg/operation/terraformer/config.go [hunk 2]
```diff
@@ -100,6 +97,12 @@ func (t *Terraformer) prepare() (int, error) {
 	return numberOfExistingResources, nil
 }
 
+// ConfigExists returns true if all three Terraform configuration secrets/configmaps exist, and false otherwise.
+func (t *Terraformer) ConfigExists() (bool, error) {
+	numberOfExistingResources, err := t.prepare()
+	return numberOfExistingResources == 3, err
```
Comment (timuthy) [inline pkg/operation/terraformer/config.go]: Would make it a constant -> Can be reused [here](https://github.com/gardener/gardener/blob/master/pkg/operation/terraformer/terraformer.go#L116)

**Review by mvladev — DISMISSED** [review]

**Review by timuthy — CHANGES_REQUESTED** [review]

File: pkg/operation/terraformer/config.go (hunk 2 above)
Comment (rfranzke) [reply pkg/operation/terraformer/config.go]: done, PTAL

File: pkg/operation/terraformer/config.go (hunk 1 above)
Comment (rfranzke) [reply pkg/operation/terraformer/config.go]: done, PTAL

**Review by timuthy — APPROVED** [review]

**Pull request merged**